pkg/cloudprovider/cloudprovider.go

```diff
@@ -98,7 +98,10 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 	}
 	nodeClassReady := nodeClass.StatusConditions().Get(status.ConditionReady)
 	if !nodeClassReady.IsTrue() {
-		return nil, fmt.Errorf("resolving ec2nodeclass, %s", nodeClassReady.Message)
+		if nodeClassReady.IsFalse() {
+			return nil, cloudprovider.NewNodeClassNotReadyError(fmt.Errorf(nodeClassReady.Message))
+		}
+		return nil, fmt.Errorf("resolving NodeClass readiness, NodeClass is in Ready=Unknown, %s", nodeClassReady.Message)
```

Comment:
nit: does it read better to treat this as a switch statement? Happy to keep as is
```
	switch state := nodeClassReady.Status; state {
	case v1.ConditionUnknown:
		return nil, fmt.Errorf("unknown")
	case v1.ConditionFalse:
		return nil, fmt.Errorf("false")
	}

```